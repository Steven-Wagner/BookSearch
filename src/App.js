import React, { Component } from 'react';
import './App.css';
import SearchInfo from './searchInfo/searchInfo';
import ListOfBooks from './books/listOfBooks/listofBooks';

export default class App extends Component {

  newSearch(term, book, print) {
    this.setState({
      searchTerm: term,
      bookType: book,
      printType: print
    })
  }

  constructor(props) {
    super(props);
    this.state= {
      currentBooks: this.props.booksData,
      searchTerm: 'q=flowers',
      bookType: '',
      printType: '&printType=all'
    }
    this.newSearch = this.newSearch.bind(this);
  }

  componentDidUpdate(prevProps, prevState) {
    console.log('update ran');
    if (prevState.searchTerm !== this.state.searchTerm || prevState.bookType !== this.state.bookType || prevState.printType !== this.state.printType) {
      const urlToGetBookData = `https://www.googleapis.com/books/v1/volumes?${this.state.searchTerm}${this.state.printType}${this.state.bookType}&key=AIzaSyDDwj1IU1Tyxthw5Y1kkKID2H6Y2Mqs6DA`;

      fetch(urlToGetBookData)
        .then(bookDataResponse => {
          if (!bookDataResponse.ok) {
            throw new Error('Somehthing went wrong with fetching from googlebooksapi')
          }
          return bookDataResponse.json();
        })
        .then(jBookResponse => {
          if (jBookResponse.items) {
            this.setState({
              currentBooks: jBookResponse.items
            })
            console.log('this is the new state of currentBooks', this.state.currentBooks);
          }
          else {
            alert('No Results Found');
          }
        })
        .catch(err => {
          console.log(err.message);
        })
    }
  }

  render() {
    return (
      <div>
        <header className="App-header">
          <h1>Google Book Search</h1>
        </header>
        <SearchInfo 
          newSearch={this.newSearch} />
        <ListOfBooks 
          currentBooks={this.state.currentBooks}/>
      </div>
    );
  }
}
