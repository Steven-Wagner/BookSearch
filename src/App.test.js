import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import testData from './testData';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App booksData={testData}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<App booksData={testData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });