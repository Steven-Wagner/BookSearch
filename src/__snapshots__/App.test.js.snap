// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <header
    className="App-header"
  >
    <h1>
      Google Book Search
    </h1>
  </header>
  <div
    className="searchInfo"
  >
    <form
      className="searchBar"
    >
      <label
        htmlFor="searchTerm"
      >
        Search:
      </label>
      <input
        id="searchTerm"
        onChange={[Function]}
        type="text"
      />
      <button
        className="submit-button"
        onClick={[Function]}
        type="submit"
      >
        Search
      </button>
    </form>
    <div
      className="searchFilters"
    >
      <div
        className="printType"
      >
        <label
          htmlFor="bookOptions"
        >
          Print Type
        </label>
        <select
          onChange={[Function]}
        >
          <option
            value="all"
          >
            All
          </option>
          <option
            value="books"
          >
            Books
          </option>
          <option
            value="magazines"
          >
            Magazines
          </option>
        </select>
      </div>
      <div
        className="bookType"
      >
        <label
          htmlFor="bookOptions"
        >
          Book Type:
        </label>
        <select
          id="bookOptions"
          name="bookOptions"
          onChange={[Function]}
        >
          <option
            value="no-filter"
          >
            No Filter
          </option>
          <option
            value="ebooks"
          >
            Ebooks
          </option>
          <option
            value="full"
          >
            Full content available
          </option>
          <option
            value="free-ebooks"
          >
            Free-ebooks
          </option>
          <option
            value="paid-ebooks"
          >
            Paid-ebooks
          </option>
        </select>
      </div>
    </div>
  </div>
  <ul
    className="listOfBooks"
  >
    <li>
      <header>
        <h2>
          The Flower Book
        </h2>
        <h3>
          
        </h3>
      </header>
      <div
        className="bookContent"
      >
        <img
          alt="The Flower Book"
          className="thumbnail"
          src="http://books.google.com/books/content?id=1hfMDQAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
        />
        <div
          className="bookInfo"
        >
          <p
            className="author-Price"
          >
            Author: 
            Rachel Siegfried
            <br />
            Price:  
            25.99
          </p>
          <p>
            The Flower Book explores 60 flowers, bloom-by-bloom in stunning portraiture
          </p>
        </div>
      </div>
    </li>
    <li>
      <header>
        <h2>
          The Language of Flowers
        </h2>
        <h3>
          A Novel
        </h3>
      </header>
      <div
        className="bookContent"
      >
        <img
          alt="The Language of Flowers"
          className="thumbnail"
          src="http://books.google.com/books/content?id=dD9_uTXl7roC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
        />
        <div
          className="bookInfo"
        >
          <p
            className="author-Price"
          >
            Author: 
            Vanessa Diffenbaugh, Steven Wagner
            <br />
            Price:  
            12.99
          </p>
          <p>
            NEW YORK TIMES BESTSELLER The Victorian language of flowers was used to convey romantic expressions: honeysuckle for devotion, asters for patience, and red roses for love
          </p>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
